### ISSUE TITLE
Gemwin only showing content in a portion of the window

### ISSUE THREAD

**Issue by b01xy** [umlaeute/Gem#437]
### Detail

Hello,

With Gem installed from Debian 12 repository, Gemwin is only showing content (from pix_film or any geo) on a portion of the window on my system.

I struggled a bit because I thought it was due to a poor opengl support of my amd rizen gpu vs nvidia prime on debian but it appeared that it was the same on linuxmint 21.3.

I found that the gemdefaultwindow in [gemwin] abstraction was [gemsdlwindow].

Changing it to [gemglxwindow] solved the problem.

Changing it to [gemglfw3window] after having installed gem-plugin-glfw3 package made the problem reappear.

I can change definitely the gemdefaultwindow to [gemglxwindow]  so for me it is solved, but is it SDL now the default rendering backend in debian ?

Many thanks !

### Gem Version

1:0.94-8+b1

### Pd Version

0.53.1

### Operating System

Linux

### OS Version

Debian GNU/Linux 12 (bookworm)

### Which CPU are you using?

amd64/x86_64 ("64bit Intel")

**Comment by umlaeute** [comment]
this seems to be a duplicate of #243 (obviously on Linux, rather than macOS).

could you confirm that the problem is fixed in the git-snapshot (available on deken, if you do not want to compile yourself)?

**Comment by b01xy** [comment]
quick answer :

yes : now it is [gemglxwindow] as default backend and it seems to work as expected
+/- no : I have a "gemglxwindow.pd_linux: undefined symbol: _ZN9CPPExtern8m_holderE" error in Pd console and there is a watchdog when I try to load the video in the patch that worked when I made the changes in Gemwin abstraction from deb repo as mentionned before, may be due to other reason ...

I'll make more test in the next days

Thanks
++

**Comment by umlaeute** [comment]
> yes : now it is [gemglxwindow] as default backend and it seems to work as expected

well, no. the deken-downloads should use `gemglfw3window` as default.
if you build Gem yourself, make sure to try with `gemglfw3window` (either with the `--with-defaultwindow=gemgflw3window` configure-flag,or by just fixing the `gemdefaultwindow.pd` abstraction)

> +/- no : I have a "gemglxwindow.pd_linux: undefined symbol: _ZN9CPPExtern8m_holderE" error in Pd console 

which indicates that you are somehow mixing "gemglxwindow.pd_linux" (e.g. from Debian) with an incompatible  "Gem.pd_linux" (e.g. from deken)

**Comment by b01xy** [comment]
>     +/- no : I have a "gemglxwindow.pd_linux: undefined symbol: _ZN9CPPExtern8m_holderE" error in Pd console
> 
> which indicates that you are somehow mixing "gemglxwindow.pd_linux" (e.g. from Debian) with an incompatible "Gem.pd_linux" (e.g. from deken)

Indeed, as a strange side effect, I was not able to load .mp4 video without a "signaling pd watchdog" and imagemagick was creating a huge magick*.pam file in /tmp/

I removed Gem from Deken and debian repository, re-installed Gem from Debian repo and changed gemdefaultwindow in [gemwin] abstraction to [gemglxwindow] via a 
`sudo pd`
to make this change permanent

So now, everything works perfectly, but before closing this issue, a last remaining question : is it on purpose to put SDL as the default rendering backend in debian package as it doesn't allow to use all the space of a Gemwin window ?

thks

**Comment by umlaeute** [comment]
> Indeed, as a strange side effect, I was not able to load .mp4 video without a "signaling pd watchdog" and imagemagick was creating a huge magick*.pam file in /tmp/

https://github.com/umlaeute/Gem/issues/297

> 
> I removed Gem from Deken and debian repository, re-installed Gem from Debian repo and changed gemdefaultwindow in [gemwin] abstraction to [gemglxwindow] via a `sudo pd` to make this change permanent

the *proper* way is to to run

```
sudo update-alternatives --config gemdefaultwindow.pd
```


> 
> So now, everything works perfectly, but before closing this issue, a last remaining question : is it on purpose to put SDL as the default rendering backend in debian package as it doesn't allow to use all the space of a Gemwin window ?

probably not.
i guess the priorities should be ajdusted (but that's a Debian bug; feel free to report it with `reportbug gem`

---

as a sidenote: i would still be interested to see whether the current `git` snapshot of Gem has this problem (with `[gemglfw3window]`!) or not.

**Comment by umlaeute** [comment]
i strongly believe that this has been fixed.
